erpnext/stock/get_item_details.py

```diff
@@ -261,6 +261,11 @@ def get_basic_details(args, item, overwrite_warehouse=True):
 		elif (args.get('doctype') in ['Purchase Order', 'Purchase Receipt', 'Purchase Invoice']) or \
 			(args.get('doctype') == 'Material Request' and args.get('material_request_type') == 'Purchase'):
 			args.uom = item.purchase_uom if item.purchase_uom else item.stock_uom
+			if args.get('doctype') == 'Purchase Invoice':
+				from erpnext.assets.doctype.asset_category.asset_category import get_asset_category_account
+				expense_account = get_asset_category_account(fieldname = "fixed_asset_account", item = args.item_code, company= args.company)
+				if not expense_account:
```

Comment:
This is not the logic. fixed_asset_account should only be used if item is a fixed asset item. otherwise it should be based on get_default_expense_account

Reply:
done